This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit 8f9763fa525b35a35d1186713bb0d59494edc4dd
Author: Tom Schuster <evilpies@gmail.com>
Date:   Sat Apr 12 23:24:49 2025 +0200

    Fix the implementation of Display and TryFrom<&str> for Teletype
    
    Fixes #7732 and fixes #7733

diff --git uucore/src/lib/features/tty.rs uucore/src/lib/features/tty.rs
index 608de78b9a..6854ba1644 100644
--- uucore/src/lib/features/tty.rs
+++ uucore/src/lib/features/tty.rs
@@ -20,9 +20,9 @@ pub enum Teletype {
 impl Display for Teletype {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
         match self {
-            Self::Tty(id) => write!(f, "/dev/pts/{}", id),
-            Self::TtyS(id) => write!(f, "/dev/tty{}", id),
-            Self::Pts(id) => write!(f, "/dev/ttyS{}", id),
+            Self::Tty(id) => write!(f, "/dev/tty{id}"),
+            Self::TtyS(id) => write!(f, "/dev/ttyS{id}"),
+            Self::Pts(id) => write!(f, "/dev/pts/{id}"),
             Self::Unknown => write!(f, "?"),
         }
     }
@@ -32,10 +32,6 @@ impl TryFrom<String> for Teletype {
     type Error = ();
 
     fn try_from(value: String) -> Result<Self, Self::Error> {
-        if value == "?" {
-            return Ok(Self::Unknown);
-        }
-
         Self::try_from(value.as_str())
     }
 }
@@ -44,6 +40,10 @@ impl TryFrom<&str> for Teletype {
     type Error = ();
 
     fn try_from(value: &str) -> Result<Self, Self::Error> {
+        if value == "?" {
+            return Ok(Self::Unknown);
+        }
+
         Self::try_from(PathBuf::from(value))
     }
 }
